### PR TITLE
use requests instead of aiohttp for underlying http client

### PR DESCRIPTION
**Pull request by SebbyLaw**
instead of starting and stopping the current thread's event loop on every http call, just make an http call.

### CODE REVIEW

File: python/lancedb/remote/client.py [hunk 1]
```diff
@@ -76,38 +76,38 @@ def headers(self) -> Dict[str, str]:
...
     @_check_not_closed
-    async def get(self, uri: str, params: Union[Dict[str, Any], BaseModel] = None):
+    def get(self, uri: str, params: Union[Dict[str, Any], BaseModel] = None):
         """Send a GET request and returns the deserialized response payload."""
         if isinstance(params, BaseModel):
             params: Dict[str, Any] = params.dict(exclude_none=True)
-        async with self.session.get(
-            uri,
+        with self.session.get(
+            self.url + uri,
```
Comment (chebbyChefNEQ) [inline python/lancedb/remote/client.py:101]: I think we should use `urllib.parse.urljoin` here

File: python/lancedb/remote/client.py [hunk 2]
```diff
@@ -76,38 +76,38 @@ def headers(self) -> Dict[str, str]:
...
             params=params,
             headers=self.headers,
-            timeout=aiohttp.ClientTimeout(total=30),
+            timeout=30,
```
Comment (chebbyChefNEQ) [inline python/lancedb/remote/client.py:104]: nit: maybe use `(5.0, 30.0)`. 5s for connection timeout and 30s for read timeout

File: python/lancedb/remote/client.py [hunk 3]
```diff
@@ -139,31 +139,26 @@ async def post(
             headers["content-type"] = content_type
         if request_id is not None:
             headers["x-request-id"] = request_id
-        async with self.session.post(
-            uri,
+        with self.session.post(
+            self.url + uri,
```
Comment (chebbyChefNEQ) [inline python/lancedb/remote/client.py:143]: ditto: use `urljoin`

**Review by chebbyChefNEQ — COMMENTED** [review]
Thank you for doing this!

I think we also need to update `pyproject.toml` to add `requests` as a dependency.

Otherwise looks good

File: python/lancedb/remote/client.py [hunk 4]
```diff
@@ -53,15 +53,15 @@ class RestfulLanceDBClient:
     closed: bool = attrs.field(default=False, init=False)
 
     @functools.cached_property
-    def session(self) -> aiohttp.ClientSession:
-        url = (
+    def session(self) -> requests.Session:
+        self.url: str = (
             self.host_override
             or f"https://{self.db_name}.{self.region}.api.lancedb.com"
         )
```
Comment (chebbyChefNEQ) [inline python/lancedb/remote/client.py:60]: let's move this to it's own property

**Comment by SebbyLaw** [comment]
> I think we also need to update `pyproject.toml` to add `requests` as a dependency.

requests is already listed as a dependency in the pyproject.toml

I've removed aiohttp from the module dependencies and replaced `requests` with it in the testing dependency extras since it is only used in tests now

**Review by chebbyChefNEQ — APPROVED** [review]
LGTM, barring CI

**Pull request merged**